src/quill-editor.component.ts

```diff
@@ -126,11 +127,11 @@ export class QuillEditorComponent implements AfterViewInit, ControlValueAccessor
 
     // update model if text changes
     this.quillEditor.on('text-change', (delta: any, oldDelta: any, source: string) => {
-      let html = this.editorElem.children[0].innerHTML;
+      let html: Html = this.editorElem.children[0].innerHTML;
```

Comment:
Quick background, if you are not using --strictNullChecks both null and undefined are in the domain of all types. so T | undefined | null is equivalent to T. if you are using --strictNullChecks however, that is not the case. T does not include null or undefined.(https://github.com/Microsoft/TypeScript/issues/8322).

So, the following code is not a valid in TS: 
     ```
 if (html === '<p><br></p>') {
        html = null;
      }
```
So, I thought use the range (String and null) created a new type "html" (Maybe the name is not the more suitable. Futhermore, I also tried your posible solution:

`let html: string? = ...`

But It is not a valid TS for my version.

In the new angular version 4.1 strictNullCheck flag come to be the default behaviour in AOT (Which is disable in the version 4.0.2).